products_application/src/App.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from 'react';
+import styles from './App.module.scss';
+import { Cards } from './components/Cards/Cards';
+import { Loader } from './components/Loader/Loader';
+import { ErrorButton } from './components/ErrorButton/ErrorButton';
+import { IAppState, IRequestResult } from './types';
+
+export class App extends Component<object, IAppState, string> {
+  state = {
+    inputValue: this.setInputValue(),
+    cards: [],
+    isLoading: true,
+    isNothingFound: false,
+    isButtonDisabled: false,
+  };
+
+  setInputValue() {
+    const keyWord = localStorage.getItem('alyona8891_keyword');
+    if (keyWord) {
+      return keyWord;
+    } else {
+      return '';
+    }
+  }
+
+  componentDidMount(): void {
+    const keyWord = localStorage.getItem('alyona8891_keyword');
+    if (keyWord) {
+      this.setState({
+        isButtonDisabled: true,
+      });
+      fetch(`https://dummyjson.com/products/search?q=${keyWord}`)
+        .then((res) => res.json())
+        .then(
+          (json: IRequestResult) => {
+            if (json.products.length === 0) {
+              this.setState({
+                isLoading: false,
+                cards: json.products,
+                isNothingFound: true,
+                isButtonDisabled: false,
+              });
+            } else {
+              this.setState({
+                isLoading: false,
+                cards: json.products,
+                isButtonDisabled: false,
+              });
+            }
+          },
+          (error: Error) => {
+            this.setState({
+              isButtonDisabled: true,
+            });
+            console.log(error);
+          }
+        );
+    } else {
+      this.setState({
+        isButtonDisabled: true,
+      });
+      fetch('https://dummyjson.com/products')
+        .then((res) => res.json())
+        .then(
+          (json: IRequestResult) => {
+            if (json.products.length === 0) {
+              this.setState({
+                isLoading: false,
+                cards: json.products,
+                isNothingFound: true,
+                isButtonDisabled: false,
+              });
+            } else {
+              this.setState({
+                isLoading: false,
+                cards: json.products,
+                isButtonDisabled: false,
+              });
+            }
+          },
+          (error: Error) => {
+            this.setState({
+              isButtonDisabled: true,
+            });
+            console.log(error);
+          }
+        );
+    }
+  }
+
+  handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (e.target) {
+      this.setState({
+        inputValue: e.target.value,
+      });
+    }
+  };
+
+  handleSearchButton = () => {
+    this.setLocalStorageData();
+    fetch(`https://dummyjson.com/products/search?q=${this.state.inputValue}`)
```

Comment:
and then, if you'll change your button to type "text" or prevent redirection, here are no handling of errors and empty result